### PR TITLE
plugins/lsp/volar: add ts-ls integration

### PR DESCRIPTION
**Pull request by khaneliman**
Resolves https://github.com/nix-community/nixvim/issues/1937

### CODE REVIEW

File: plugins/lsp/language-servers/default.nix [hunk 1]
```diff
@@ -691,6 +691,32 @@ let
...
+          description = ''
+            Enable integration with TypeScript language server.
+          '';
+          default = true;
+          example = false;
+        };
+      };
+      extraConfig = cfg: {
+        plugins.lsp.servers.ts-ls = {
+          filetypes = [ "vue" ];
```
Comment (traxys) [inline plugins/lsp/language-servers/default.nix:706]: If we don't have a matching default for ts files this will disable them won't it?

Comment (khaneliman) [reply plugins/lsp/language-servers/default.nix:706]: This block was intended to just append to the user's existing ts-ls config when the integration is enabled.

Comment (khaneliman) [reply plugins/lsp/language-servers/default.nix:706]: Added a couple tests to show what I meant.

Comment (traxys) [reply plugins/lsp/language-servers/default.nix:706]: The issue is that `ts-ls.filetypes` is `null` by default. This means that enabling the `tslsIntegration` but leaving the `ts-ls.filetypes` as default will result on only  `{ "vue" }` being passed to nvim-lspconfig. I think we need to set the propagate the default value of  `{ "javascript", "javascriptreact", "javascript.jsx", "typescript", "typescriptreact", "typescript.tsx" }`

Comment (khaneliman) [reply plugins/lsp/language-servers/default.nix:706]: Oh, you mean the plugin's default has those values and we override the default plugin behavior by enabling it.

Comment (traxys) [reply plugins/lsp/language-servers/default.nix:706]: Exactly, I think the fix is to hardcode the plugin's default as the nix default

Comment (khaneliman) [reply plugins/lsp/language-servers/default.nix:706]: Sounds good, I'll update that when I get home later.

**Review by traxys — APPROVED** [review]

**Comment by khaneliman** [comment]
@mergifyio queue

**Pull request merged**